File: CapstoneProject1/src/main/java/SeleniumTestNG/Pages/SearchResultsPage.java
package SeleniumTestNG.Pages;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class SearchResultsPage {
	private WebDriver driver;
	WebElement firstElement; 

    private By firstItem = By.xpath("//div[contains(@class, 'KzDlHZ')]");

    public SearchResultsPage(WebDriver driver) {
        this.driver = driver;
    }
    public String FirstItemName() {
    	//Added logic to pick first item displayed on the screen
    	List<WebElement> elements = driver.findElements(firstItem);
        if (!elements.isEmpty()) {
             firstElement = elements.get(0);
             return firstElement.getText();
        } else {
        	System.out.println("Element Not found!");
            return null;
        }
    }

    public void clickFirstItem() { 	
    	firstElement.click();
    	
    }
    

}

File: Python_Program/pages/search_results_page.py
from selenium.webdriver.common.by import By
from pages.base_page import BasePage

class SearchResultsPage(BasePage):
    FIRST_ITEM = (By.XPATH, "(//div[@class='_4rR01T' or @class='s1Q9rs'])[1]")

    def click_first_item(self):
        self.click(self.FIRST_ITEM)
